src/active_item_cache.h

```diff
@@ -38,22 +39,23 @@ struct hash<special_item_type> {
         return static_cast<size_t>( k );
     }
 };
+
+template<>
+struct hash<safe_reference<item>> {
+    std::size_t operator()( safe_reference<item > const &s ) const noexcept {
+        return hash<item * > {}( s.get() );
```

Comment:
The same safe reference object can generate different hash values before and after the item has been destroyed.